rust/main/agents/relayer/src/msg/pending_message.rs

```diff
@@ -664,6 +674,26 @@ impl PendingMessage {
             }
         }))
     }
+
+    async fn clarify_reason(&self, reason: ReprepareReason) -> ReprepareReason {
+        use hyperlane_application::ApplicationOperationVerifierError::*;
+        use ReprepareReason::ApplicationOperationVerificationFailed;
+
+        match self
+            .ctx
+            .application_operation_verifier
+            .verify(&self.app_context, &self.message)
+            .await
+        {
+            Ok(()) => reason,
+            Err(e) => match e {
+                InsufficientAmountError | MalformedMessageError(_) => {
+                    ApplicationOperationVerificationFailed
```

Comment:
I think it'd be useful to include the error type in the label btw, such as `AppVerificationFailed(InsufficientAmount)` or `AppVerificationFailed(MalformedMessageError)`

Reply:
It will create circular dependency, so, we won't do it.